### PR TITLE
internal/envconfig: add convenience boolFromEnv to improve readability

### PR DESCRIPTION
**Pull request by dfawley**
RELEASE NOTES: none

### CODE REVIEW

File: internal/envconfig/envconfig.go [hunk 1]
```diff
@@ -25,25 +25,26 @@ import (
...
 	// checking which NACKs configs specifying ring sizes > 8*1024*1024 (~8M).
 	RingHashCap = uint64FromEnv("GRPC_RING_HASH_CAP", 4096, 1, 8*1024*1024)
 )
 
+func boolFromEnv(envVar string, def bool) bool {
+	if def {
+		return !strings.EqualFold(os.Getenv(envVar), "false")
+	}
+	return strings.EqualFold(os.Getenv(envVar), "true")
+}
```
Comment (easwars) [inline internal/envconfig/envconfig.go:46]: Would it be better to use `os.LookupEnv()` here? I feel this is a little easier to read. What do you think?

```golang
func boolFromEnv(envVar string, def bool) bool {
	val, ok := os.LookupEnv(envVar)
	if !ok {
		return def
	}
	return strings.EqualFold(val, "true")
}
```

Comment (dfawley) [reply internal/envconfig/envconfig.go:46]: This isn't right, though?  The default value affects whether we require "true" or "false" to go against the default.  I.e. if the default is `true` then we will return `true` from this function unless the env var contains `"false"` (case-insensitive).

Comment (easwars) [reply internal/envconfig/envconfig.go:46]: Could we add a docstring clarifying this. Thanks.

**Review by easwars — APPROVED** [review]
LGTM modulo the docstring.

File: internal/envconfig/envconfig.go (hunk 1 above)
Comment (dfawley) [reply internal/envconfig/envconfig.go:46]: Done!

**Pull request merged**